packages/core/contracts/data-verification-mechanism/test/MockOracleAncillary.sol

```diff
@@ -79,22 +88,31 @@ contract MockOracleAncillary is OracleAncillaryInterface, Testable {
         bytes memory ancillaryData,
         int256 price
     ) public {
-        verifiedPrices[identifier][time][ancillaryData] = Price(true, price, getCurrentTime());
+        bytes32 requestId = _encodePriceRequest(identifier, time, ancillaryData);
+        verifiedPrices[requestId] = Price(true, price, getCurrentTime());
 
-        QueryIndex storage queryIndex = queryIndices[identifier][time][ancillaryData];
+        QueryIndex storage queryIndex = queryIndices[requestId];
         require(queryIndex.isValid, "Can't push prices that haven't been requested");
         // Delete from the array. Instead of shifting the queries over, replace the contents of `indexToReplace` with
         // the contents of the last index (unless it is the last index).
         uint256 indexToReplace = queryIndex.index;
-        delete queryIndices[identifier][time][ancillaryData];
+        delete queryIndices[requestId];
         uint256 lastIndex = requestedPrices.length - 1;
         if (lastIndex != indexToReplace) {
             QueryPoint storage queryToCopy = requestedPrices[lastIndex];
-            queryIndices[queryToCopy.identifier][queryToCopy.time][queryToCopy.ancillaryData].index = indexToReplace;
+            queryIndices[_encodePriceRequest(queryToCopy.identifier, queryToCopy.time, queryToCopy.ancillaryData)]
+                .index = indexToReplace;
             requestedPrices[indexToReplace] = queryToCopy;
         }
+        requestedPrices.pop();
```

Comment:
This was a bug in previous implementation and caused `getPendingQueries()` to return price requests that were resolved.